frame/contracts/src/exec.rs

```diff
@@ -43,13 +43,82 @@ pub type StorageKey = [u8; 32];
 /// A type that represents a topic of an event. At the moment a hash is used.
 pub type TopicOf<T> = <T as frame_system::Config>::Hash;
 
-/// Describes whether we deal with a contract or a plain account.
-pub enum TransactorKind {
-	/// Transaction was initiated from a plain account. That can be either be through a
-	/// signed transaction or through RPC.
-	PlainAccount,
-	/// The call was initiated by a contract account.
-	Contract,
+/// Information needed for rent calculations that can be requested by a contract.
+#[derive(codec::Encode)]
+#[cfg_attr(test, derive(Debug, PartialEq))]
+pub struct RentParams<T: Config> {
+	/// The total balance of the contract. Includes the balance transferred from the caller.
+	total_balance: BalanceOf<T>,
+	/// The free balance of the contract. Includes the balance transferred from the caller.
+	free_balance: BalanceOf<T>,
+	/// See crate [`Contracts::subsistence_threshold()`].
+	subsistence_threshold: BalanceOf<T>,
+	/// See crate [`Config::DepositPerContract`].
+	deposit_per_contract: BalanceOf<T>,
```

Comment:
I couldn't understand this from the referenced docs: Does `DepositPerContract` include `TombstoneDeposit`? I suppose not, meaning that the total balance a contract needs to stay alive would be `DepositPerContract + TombstoneDeposit`?

This would contradict the docs for `DepositPerContract`:
```
The balance every contract needs to deposit to stay alive indefinitely.
```

Reply:
Those two are independent. The `subsistence_threshold` is the existential deposit for contracts: You cannot go below it without using `seal_terminate` (the supervisor enforces this). The only interaction with the rent system is that it can prevent making a rent payment which results in the creation of the tombstone rather than making the payment. If you are merely enforcing that you never pay rent by having enough balance at all times this value is of no interest to you.

`deposit_per_contract` on the other hand determines how much you have to deposit per contract in order to stop paying rent given a contract which neither has any code (not possible) nor data storage.